handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -230,42 +230,44 @@ protected void deallocate() {
         this.jdkCompatibilityMode = jdkCompatibilityMode;
         Lock readerLock = context.ctxLock.readLock();
         readerLock.lock();
+        final long ssl;
         try {
-            ssl = SSL.newSSL(context.ctx, !context.isClient());
+            this.ssl = ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
@johnou Is that even possible?  Destroyed is only set from shutdown, which shouldn't be possible to invoke before the Ctor has finished.  That is, unless SSL.newSSL somehow had a reference back to the instance which has yet to be constructed.